src/kafkit/registry/manager.py

```diff
@@ -0,0 +1,312 @@
+      compatibility setting.
+
+    - Subjects follow the **record name** naming strategy
+      (``RecordNameStrategy``). That is, the subject name is the
+      fully-qualified name of the Avro Schema.
+
+      *Note that this is different from the default naming strategy,*
+      ``TopicNameStrategy``, where subjects are named for the topic with
+      ``-key`` or ``-value`` suffixes.
+
+    For more information, see :doc:`/recordnameschemamanager-howto` in
+    the user guide.
+    """
+
+    def __init__(
+        self, *, root: Path, registry: RegistryApi, suffix: str = ""
+    ) -> None:
+        self._logger = logging.getLogger(__name__)
+        self._root = root
+        self._registry = registry
+        self._suffix = suffix
+
+        self._serializer = PolySerializer(registry=self._registry)
+        self.schemas: Dict[str, Any] = {}
+
+        self._load_schemas()
+
+    def _load_schemas(self) -> None:
+        """Load all schemas found in the root directory into the instance
+        storage.
+        """
+        schema_paths = self._root.rglob("*.json")
+        for schema_path in schema_paths:
+            schema = json.loads(schema_path.read_text())
+
+            if self._suffix:
+                schema["name"] = f'schema["name"]{self._suffix}'
+
+            fqn = self._get_fqn(schema)
+            self.schemas[fqn] = schema
+
+    @staticmethod
+    def _get_fqn(schema: Mapping[str, Any]) -> str:
+        """Get the fully-qualified name of an Avro schema.
+
+        Parameters
+        ----------
+        schema : `dict`
+            The Avro schema.
+
+        Returns
+        -------
+        str
+            The fully-qualified name.
+
+        Notes
+        -----
+        The decision sequence is:
+
+        - If the ``name`` field includes a period (``.``), the ``name`` field
+          is treated as a fully-qualified name.
+        - Otherwise, if the schema includes a ``namespace`` field, the
+          fully-qualified name is ``{{namespace}}.{{name}}``.
+        - Otherwise, the ``name`` is treated as the fully-qualified name.
+        """
+        if "." not in schema["name"] and "namespace" in schema:
+            fqn = ".".join((schema["namespace"], schema["name"]))
+        else:
+            fqn = schema["name"]
+        assert isinstance(fqn, str)
+        return fqn
+
+    async def register_schemas(
+        self, *, compatibility: Optional[str] = None
+    ) -> None:
+        """Register all local schemas with the Confluent Schema Registry.
+
+        Parameters
+        ----------
+        compatibility : `str`, optional
+            The compatibility setting to apply to all subjects. Allowed values:
+
+            - ``"BACKWARD"``
+            - ``"BACKWARD_TRANSITIVE"``
+            - ``"FORWARD"``
+            - ``"FORWARD_TRANSITIVE"``
+            - ``"FULL"``
+            - ``"FULL_TRANSITIVE"``
+            - ``"NONE"``
+            - `None`
```

Comment:
I know upstream decides on those names and takes a `str`, but since you're verifying the parameter is valid anyway, consider making this an enum so that you get some type checking and typo protection earlier than at runtime.